### PR TITLE
Add support for custom column mapping

### PR DESCRIPTION
**Pull request by raymondfeng**
The PR adds the support for mssql column mapping for model properties.

/to @ritch or @bajtos

### CODE REVIEW

File: test/mapping.test.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            'columnName': 'PRODUCT_ID',
+            'dataType': 'uniqueidentifier',
+            'nullable': 'N'
+          }
+        },
+        'locationId': {
+          'type': 'String', 'required': true, 'length': 20,
+          'mssql': {
+            'columnName': 'LOCATION_ID', 'dataType': 'nvarchar',
+            'nullable': 'N'
```
Comment (bajtos) [inline test/mapping.test.js]: I find this style confusing. IMO you should either put all items on a single line, or else put each key/value pair on a new line.

File: test/mapping.test.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          Model.create({locationId: 'l001',
+            available: 10, total: 50}, callback);
+        },
+        function(callback) {
+          Model.create({locationId: 'l002',
+            available: 30, total: 40}, callback);
+        },
+        function(callback) {
+          Model.create({locationId: 'l001',
+            available: 15, total: 30}, callback);
```
Comment (bajtos) [inline test/mapping.test.js]: I find this style confusing. IMO you should either put all key/value pairs on a single line, or else put each key/value pair on a new line.

```js
Model.create(
  { locationId: 'l001', available: 15, total: 30 },
  callback);
```

The same comment applies to L66-L67 and L70-L71.

**Comment by bajtos** [comment]
I am not familiar with this code base. A superfluous inspection did not reveal any major issues. My main concern is whether the new tests cover all places you had to change? See also my style-related comments above.

I'll leave the ultimate decision up to you, feel free to ignore my comments and merge the PR as it is now.

**Comment by paulomcnally** [comment]
This solved relations?

With not custom column mapping relations don't work.

**Comment by paulomcnally** [comment]
Please send release to npm. :+1:

**Pull request merged**